quinn-proto/src/endpoint.rs

```diff
@@ -1059,10 +1167,27 @@ impl fmt::Debug for Incoming {
             // rest is too big and not meaningful enough
             .field("retry_src_cid", &self.retry_src_cid)
             .field("orig_dst_cid", &self.orig_dst_cid)
+            .field("incoming_idx", &self.incoming_idx)
+            // improper drop warner contains no information
             .finish_non_exhaustive()
     }
 }
 
+struct IncomingImproperDropWarner;
```

Comment:
By destructuring `Incoming` at its consumption site, can we implement the warning directly on `Incoming::drop()`? If not, can we use destructuring instead of `mem::forget()` inside `IncomingImproperDropWarner::dismiss()`?

Reply:
Types which implement drop cannot have their ownership taken by destructuring

```rs
// this compiles:
struct A(Box<i32>);
fn a(a: A) {
    let A(i) = a;
}

// this does not compile:
/*error[E0509]: cannot move out of type `B`, which implements the `Drop` trait
  --> src/lib.rs:12:16
   |
12 |     let B(i) = b;
   |           -    ^ cannot move out of here
   |           |
   |           data moved here
   |           move occurs because `i` has type `Box<i32>`, which does not implement the `Copy` trait
   |
help: consider borrowing the pattern binding
   |
12 |     let B(ref i) = b;
   |           +++*/
struct B(Box<i32>);
impl Drop for B { fn drop(&mut self) {} }
fn b(b: B) {
    let B(i) = b;
}
```

(This is what you're referring to, right?)

Reply:
Ah yes -- thanks for trying!